backend/test/Test/UploadSpec.hs

```diff
@@ -101,6 +107,19 @@ spec = withContext $ startWithInitial $ do
                   , mrIsEnabled = False
                   }
         onlyDisabled secondaryMeasurements `shouldBe` secondaryDisabled
+
+    it "rollbacks targets and compounds due processing error" $ \ctx -> do
+      runRIO ctx (uploadFile' (SqlId 1) (SqlId 1) "" (error "e") "" sampleFile)
+        `shouldThrow` anyErrorCall
+
+      targets <- runRIO ctx $ Library.getTargets Nothing Nothing Nothing
+      compounds <- runRIO ctx $ Library.getCompounds Nothing Nothing Nothing
+      experiments <- runRIO ctx $ Dashboard.getExperiments Nothing Nothing Nothing
+
+      -- We do not use 0 here, because we add some data in upper tests
```

Comment:
Actually it's trickier, you shouldn't rely on which tests will run before. For example if you pass `--match rollback` to the test-suite, this test will fail because it will run only this test. When I was writing such tests I tried to carefully structure them, so that no matter what pattern we pass to `--match`, all tests will succeed. Specifically, it's better to put tests that don't mutate DB before mutating tests. In this case it should solve the problem, just move this test before `"successfully adds targets and compounds"` and test that all numbers below are 0.

If you do it, you can also define ```shouldBeZero l = liftIO $ length l `shouldBe` 0``` and use `runRIO ctx` once instead of 3 times:
```
runRIO ctx $ do
  Library.getTargets Nothing Nothing Nothing >>= shouldBeZero
  Library.getCompounds Nothing Nothing Nothing >>= shouldBeZero
  Dashboard.getExperiments Nothing Nothing Nothing >>= shouldBeZero
```

Reply:
That was first I've did. But apparently, after transaction rollback postgres increases auto_increment index.
I can put it in separate describe section with `startWithInitial`

Reply:
executing `_ <- ctx <$ runWithInit ctx addInitialData` before other actions inside tests works as well

Reply:
Oh, indeed, there is a problem. Okay, `runWithInit ctx addInitialData` looks good to me.